server.go

```diff
@@ -21,6 +21,25 @@ import (
 // Public types
 //
 
+// DoubleSlashFixHandler is a specialized handler that wraps an HTTP mux and
+// deduplicates any doubled slashes that are included in an incoming path. So
+// `//v1/charges` would become `/v1/charges`. This works around the standard Go
+// behavior, which is to redirect the request with a 301 before it reaches the
+// underlying handler.
+//
+// The reason we deduplicate is that in some API libraries occasionally
+// generate paths with double slashes, and the real Stripe API responds to
+// these requests normally, so stripe-mock emulates that behavior.
+type DoubleSlashFixHandler struct {
+	Mux http.Handler
+}
+
+// ServeHTTP serves an HTTP request.
+func (h *DoubleSlashFixHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	r.URL.Path = strings.Replace(r.URL.Path, "//", "/", -1)
```

Comment:
I don't feel super strongly about it, but maybe we should collapse any number of consecutive slashes into a single one? Something like this:

```go
re := regexp.MustCompile(`/{2,}`)
r.URL.Path = re.ReplaceAllString(r.URL.Path, "/")
```

Reply:
Thanks OB.

So this is kind of crazy, but while the API will take doubled-slashes, it actually won't take tripled slashes. e.g.

```
$ curl 'https://api.stripe.com//v1/customers/cus_FU5uUUsu4AU3k1' -u sk_test_xxx:
{
  "id": "cus_FU5uUUsu4AU3k1",
  "object": "customer",
  "account_balance": 0,
...

$ curl 'https://api.stripe.com///v1/customers/cus_FU5uUUsu4AU3k1' -u sk_test_xxx: -i
HTTP/2 404
server: nginx
date: Mon, 22 Jul 2019 22:59:24 GMT
content-length: 0
strict-transport-security: max-age=31556926; includeSubDomains; preload
```

So I figure we should just stick to a single deduplication pass to match that behavior.

Another pretty good reason is that I think the most common cause of a duplication is by appending two string variables together where you forget that you have both a trailing slash on the first as well as a leading slash on the other:

```
s1 = "https://api.stripe.com/"
s2 = "/v1/charges"
s1 + s2
```

That's pretty easy to do by accidentally, but getting a third slash in there is a little harder.

Reply:
Haha I didn't think to test it and just assumed whatever normalization we did would work for any number of slashes. 👍 to sticking to the API's behavior.